lib/elm-test.js

```diff
@@ -444,3 +324,123 @@ ${testFileGlobs.join('\n')}
 Are the above patterns correct? Maybe try running elm-test with no arguments?
       `.trim();
 }
+
+function checkNodeJsVersion() {
+  const nodeVersionMin = '10.13.0';
+  const nodeVersionString = process.versions.node;
+
+  if (
+    nodeVersionString.localeCompare(nodeVersionMin, 'en', { numeric: true }) < 0
+  ) {
+    console.error(`You are using Node.js v${nodeVersionString}.`);
+    console.error(
+      `elm-test requires Node.js v${nodeVersionMin} or greater - upgrade the installed version of Node.js and try again!`
+    );
+    process.exit(1);
+  }
+}
+
+function main() {
+  process.title = 'elm-test';
+
+  checkNodeJsVersion();
+
+  const parseResult = Flags.parseArgv(process.argv.slice(2));
+  if (parseResult.tag === 'Error') {
+    console.error(parseResult.error);
+    throw process.exit(1);
+  }
+  const parsed = parseResult.value;
+
+  const processes = Math.max(1, os.cpus().length);
+
+  const requireElmJsonFile = () => {
+    if (!fs.existsSync('elm.json')) {
+      console.error(
+        `\`elm-test ${parsed.command.tag}\` must be run in the same directory as an existing elm.json file! To make one: elm init`
+      );
+      throw process.exit(1);
+    }
+  };
+
+  switch (parsed.command.tag) {
```

Comment:
Does flow check this `switch` is exhaustive?

Reply:
👆

Reply:
[`02aa8c2` (#465)](https://github.com/rtfeldman/node-test-runner/pull/465/commits/02aa8c2f38aa0e8dedd1375fb180c6b170ce631d)

Reply:
👍